gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/EnforceFlowFinishDeadlineDagProc.java

```diff
@@ -41,8 +44,18 @@ public EnforceFlowFinishDeadlineDagProc(EnforceFlowFinishDeadlineDagTask enforce
     super(enforceFlowFinishDeadlineDagTask);
   }
 
-  protected void enforceDeadline(DagManagementStateStore dagManagementStateStore, Dag<JobExecutionPlan> dag)
-      throws IOException {
+  @Override
```

Comment:
i see some opportunity to de-deuplicate code here
you can move this to parent class and have both the child classes implement abstract method `getDagActionType`

Reply:
good idea, done

Reply:
was able to get the action type from dagTask